pytensor/tensor/math.py

```diff
@@ -3043,6 +3043,65 @@ def vectorize_node_dot_to_matmul(op, node, batched_x, batched_y):
         return vectorize_node_fallback(op, node, batched_x, batched_y)
 
 
+def nan_to_num(x, nan=0.0, posinf=None, neginf=None):
+    """
+    Replace NaN with zero and infinity with large finite numbers (default
+    behaviour) or with the numbers defined by the user using the `nan`,
+    `posinf` and/or `neginf` keywords.
+
+    NaN is replaced by zero or by the user defined value in
+    `nan` keyword, infinity is replaced by the largest finite floating point
+    values representable by ``x.dtype`` or by the user defined value in
+    `posinf` keyword and -infinity is replaced by the most negative finite
+    floating point values representable by ``x.dtype`` or by the user defined
+    value in `neginf` keyword.
+
+    Parameters
+    ----------
+    x : symbolic tensor
+        Input array.
+    nan
+        The value to replace NaN's with in the tensor (default = 0).
+    posinf
+        The value to replace +INF with in the tensor (default max
+        in range representable by ``x.dtype``).
+    neginf
+        The value to replace -INF with in the tensor (default min
+        in range representable by ``x.dtype``).
+
+    Returns
+    -------
+    out
+        The tensor with NaN's, +INF, and -INF replaced with the
+        specified and/or default substitutions.
+    """
+    # Replace NaN's with nan keyword
+    is_nan = isnan(x)
+    is_pos_inf = eq(x, np.inf)
+    is_neg_inf = eq(x, -np.inf)
```

Comment:
Can also add the numpy helpers for `isneginf`, `isposinf` for users (and use them here)

Reply:
I created ```Isposinf``` and ```Isneginf``` with the very same purpose here:
https://github.com/Dhruvanshu-Joshi/pytensor/commit/829309bcccdf5cc8b67c76f2230d4dfd3c3fda3c

How can we use ```np.isposinf(x)``` directly in helper functions without creating an ```op``` when x is but a container?

Reply:
You don't, you use the `pt.isponsinf` helper which will have this exact code inside `pt.eq(pt.as_tensor(x), np.inf)`.